template.c

```diff
@@ -485,7 +485,7 @@ doEscape(uchar **pp, rs_size_t *pLen, unsigned short *pbMustBeFreed, int mode)
 		} else if((mode == SQL_ESCAPE) && *p == '\\') {
 			CHKiRet(cstrAppendChar(pStrB, '\\'));
 			iLen++;	/* reflect the extra character */
-		} else if((mode == JSON_ESCAPE) && (*p != '"' || *p != '\' )) {
+		} else if((mode == JSON_ESCAPE) && (*p != '"' || *p != '\\' )) {
```

Comment:
IMHO this predicte is wrong and will lead to all chars being escaped. The testbench doesn't catch that, because it does not yet have a test for this functionality (will add one).

Think about it: when *p is '"', it is not '\', so the OR part will always be true. Some for *p not in { '"', '\'}.